src/core/cast/raycast.template.js

```diff
@@ -24,6 +24,8 @@ function _raycast( nodeIndex32, bvh, side, ray, intersects ) {
 
 		/* @if INDIRECT */
 
+		// It might happen that a triangle less distant than 'near' is selected here. Should we add the near check here too? is it worth it? It might slow down to handle really rare cases
```

Comment:
It might happen that a triangle less distant than 'near' is selected here. Should we add the near check here too? is it worth it? It might slow down to handle really rare cases

Reply:
The triangle hits will also need to be filtered so that the results returned by the raycast function are accurate. Otherwise its the case that we may have 10 triangles in a large leaf node such that the near hit point falls within the ray range but the triangles within do not.

We could filter them afterward but it probably makes sense to just not add them to the intersections array here.

Reply:
It sounds like we can remove this line now, too?